nomad/structs/services.go

```diff
@@ -1352,9 +1352,12 @@ func (p *ConsulGatewayProxy) Copy() *ConsulGatewayProxy {
 		return nil
 	}
 
-	bindAddresses := make(map[string]*ConsulGatewayBindAddress, len(p.EnvoyGatewayBindAddresses))
-	for k, v := range p.EnvoyGatewayBindAddresses {
-		bindAddresses[k] = v.Copy()
+	var bindAddresses map[string]*ConsulGatewayBindAddress
```

Comment:
I'm not completely sure how the changes to setJobStatus ended up impacting or uncovering this nil vs empty bug but this fixes the failing job_endpoint_test spec

Reply:
Is there any risk that not initializing this map will cause us a NPE later? (ex. https://play.golang.org/p/tah4DfaFSy0)

Reply:
Yes, I suspect the `job.SpecChanged` may need to use something other than `reflect.DeepEqual` for equality

Reply:
I guess I'm suggesting we could initialize it as empty. It's not very costly to do.

Reply:
Yeah the empty map should be fine. This section of the spec has been awkward and the source of several NPEs already - there's multiple "defaults" depending on other conditions so I left the canonical to be nil for a lot of things. Which can be bad.

Reply:
Addressed in https://github.com/hashicorp/nomad/pull/9768/commits/29938df96b2987aa0ab205a3e67b9dbe00b15b0c